python/mlx/nn/layers/pooling.py

```diff
@@ -0,0 +1,155 @@
+        self.mode = mode
+
+    def _extra_repr(self):
+        return f"{self.kernel_size}, stride={self.stride}, padding={self.padding}, mode={self.mode}"
+
+    def _get_padding(self, features_sizes: List[int]) -> List[int]:
+        if isinstance(self.padding, int):
+            return (
+                [(0, 0)]
+                + [(self.padding, self.padding)] * len(features_sizes)
+                + [(0, 0)]
+            )
+        if len(self.padding) != len(features_sizes):
+            raise ValueError(
+                "the number of provided padding values must match the number of feature axes"
+            )
+        return [(0, 0)] + [(p, p) for p in self.padding] + [(0, 0)]
+
+    def _get_strides(self, features_sizes: List[int]) -> List[int]:
+        if isinstance(self.stride, int):
+            return [self.stride] * len(features_sizes)
+        if len(self.stride) != len(features_sizes):
+            raise ValueError(
+                "the number of provided strides must match the number of feature axes"
+            )
+        return self.stride
+
+    def _get_kernel_size(self, features_sizes: List[int]) -> List[int]:
+        if isinstance(self.kernel_size, int):
+            return [self.kernel_size] * len(features_sizes)
+        if len(self.kernel_size) != len(features_sizes):
+            raise ValueError("kernel_size must match the number of feature axes")
+        return self.kernel_size
+
+    def _pad(self, a: mx.array) -> mx.array:
+        feature_size = a.shape[1:-1]
+        padding_value = {"max": float("-inf"), "mean": 0}[self.mode]
+        return mx.pad(a, self._get_padding(feature_size), padding_value)
+
+    def __call__(self, a: mx.array) -> mx.array:
+        if a.ndim < 2:
+            raise ValueError("the input must be at least two-dimensional.")
+        # Select pooling operator
+        pool = {"max": mx.max, "mean": mx.mean}[self.mode]
+        # Pad if necessary
+        if self.padding != 0:
+            a = self._pad(a)
+        # Assumes a.shape = (batch_size, ..., num_channels)
+        batch_size, batch_stride = a.shape[0], a.strides[0]
+        feature_size, feature_strides = a.shape[1:-1], a.strides[1:-1]
+        num_channels, channels_stride = a.shape[-1], a.strides[-1]
+        # Get kernel size and strides
+        kernel_size = self._get_kernel_size(feature_size)
+        strides = self._get_strides(feature_size)
+        # Compute windows
+        windows_shape = tuple(
+            [batch_size]
+            + [
+                (f - k) // s + 1
+                for (f, k, s) in zip(feature_size, kernel_size, strides)
+            ]
+            + kernel_size
+            + [num_channels]
+        )
+        windows_strides = tuple(
+            [batch_stride]
+            + [s * f_s for (s, f_s) in zip(strides, feature_strides)]
+            + feature_strides
+            + [channels_stride]
+        )
+        windows = mx.as_strided(a, windows_shape, windows_strides)
```

Comment:
In MLX's `as_strided` the provided shape, strides and offset should be provided as if the original array is row contiguous. This is also why we don't need access to the original strides.

Reply:
Assuming we cannot expose ``array.strides()``, how would you access them to compute``windows_strides``? 
Given the convention that ``strides`` should be provided as if the original array is row contiguous, we could manually compute strides based on datatype byte size and the shape. Is there a better way?